src/psyclone/dynamo0p3.py

```diff
@@ -6285,133 +7348,84 @@ def __init__(self, kern, parent):
                 "is not yet supported for inter-grid kernels".
                 format(kern.name))
 
-        parent.add(UseGen(parent, name="constants_mod", only=True,
-                          funcnames=["r_def"]))
         self._first_arg = True
         self._first_arg_decl = None
         ArgOrdering.__init__(self, kern)
-        self._parent = parent
         self._arglist = []
         self._name_space_manager = NameSpaceFactory().create()
 
     def cell_position(self):
-        ''' Add cell position to the argument list if required '''
-        from psyclone.f2pygen import DeclGen
+        ''' Add cell position to the argument list if required. '''
```

Comment:
No need for "if required" here and below (mesh_height, mesh_ncell2d), as it adds it whether you need it or not if you call it!

Reply:
Good point.